### ISSUE TITLE
Do you plan to support time-based assertions ?

### ISSUE THREAD

**Issue by legzo** [Orange-OpenSource/hurl#85]
For some use-cases we would like to check that the http response has been returned within x milliseconds, do you think it has a chance to be implemented ? Maybe we could think about the response payload size also (if `Content-Length` header is not set correctly)

What are your thoughts ?

**Comment by fabricereix** [comment]
This is clearly a useful assertion.

Currently the response time can be displayed in verbose mode
```
$ echo GET http://localhost:8000/hello | hurl -v
* fail fast: true
* insecure: false
* follow redirect: false
* max redirect: 50
* ------------------------------------------------------------------------------
* executing entry 1
* 
* Cookie store:
* 
* Request
* GET http://localhost:8000/hello
* 
> GET /hello HTTP/1.1
> Host: localhost:8000
> Accept: */*
> User-Agent: hurl/0.99.14
> 
< HTTP/1.0 200 OK
< Content-Type: text/html; charset=utf-8
< Content-Length: 12
< Server: Flask Server
< Date: Wed, 18 Nov 2020 09:38:26 GMT
< 
* Response Time: 6ms
* 
Hello World!
```
and the json report
```

echo GET http://localhost:8000/hello | hurl --json /tmp/test.json
$ cat /tmp/test.json
[
  {
    "filename": "-",
    "entries": [
      {
        "request": {
          "method": "GET",
          "url": "http://localhost:8000/hello",
          "queryString": [],
          "headers": [],
          "cookies": [],
          "multipartFormData": [],
          "body": ""
        },
        "response": {
          "httpVersion": "HTTP/1.0",
          "status": 200,
          "cookies": [],
          "headers": [
            {
              "name": "Content-Type",
              "value": "text/html; charset=utf-8"
            },
            {
              "name": "Content-Length",
              "value": "12"
            },
            {
              "name": "Server",
              "value": "Flask Server"
            },
            {
              "name": "Date",
              "value": "Wed, 18 Nov 2020 09:32:31 GMT"
            }
          ]
        },
        "captures": [],
        "asserts": [],
        "time": 6
      }
    ],
    "success": true,
    "time": 6,
    "cookies": []
  }
```

**Comment by jcamiel** [comment]
Yes good idea !

One can use, as a current alternative,  [max-time option](https://hurl.dev/docs/man-page.html#max-time) (specified in seconds...) but it could be cool to support it per request as an assert.

One possible format: add a new query for duration (similar to status query)

```
GET http://sample.com
HTTP/1.1 200
[Asserts]
maxDuration equals 400
```

Or extend predicate with greaterThan, lessThan

```
GET http://sample.com
HTTP/1.1 200
[Asserts]
duration lessThan 400
```

**Comment by fabricereix** [comment]
I prefer a lot more the second one.
duration is naturally attached to the entry (time in the json report).
Comparison predicates (such as lessThan) can also be easily reused.

**Comment by jcamiel** [comment]
Yes and a `duration` query is more natural with captures 

ˋˋˋ
GET http://sample.net
HTTP/* 200
[Captures]
my_var: duration
ˋˋˋ

So if we add `greaterThan`, and ˋlessThan`, we should also add `countGreaterThan` and `countLessThan`.

**Comment by legzo** [comment]
Nice ! Thanks

**Comment by fabricereix** [comment]
This will be implemented in the next release 1.0.0, features #89 and #90
